src/Groups/GAPGroupsLibrary.jl

```diff
@@ -0,0 +1,476 @@
+      end
+   pos+=1
+   end
+   if isargument L1[length(L1)]=true end
+   L1 = GAP.julia_to_gap(L1)
+
+   K = GAP.Globals.CallFuncList(GAP.Globals.AllTransitiveGroups,L1)
+   return [PermGroup(K[i]) for i in 1:length(K)]          # GAP.julia_to_gap(K) does not work
+end
+
+###################################################################
+# Perfect groups
+###################################################################
+
+"""
+    perfect_group(n::Int, i::Int)
+    perfect_group(::Type{T}, n::Int, i::Int)
+Return the `i`-th group of order `n` and type `T` in the catalogue of perfect groups in the GAP Small Groups Library. The type `T` can be either ``PermGroup`` or ``FPGroup``. The group is given of type ``FPGroup`` if `T` is not specified.
+"""
+function perfect_group(::Type{T}, n::Int, m::Int) where T <: GAPGroup
+   @assert m<= number_perfect_groups(n) "There are less than $m perfect groups of order $n."
+   if T==PermGroup
+      G = T(GAP.Globals.PerfectGroup(GAP.Globals.IsPermGroup,n,m))
+   elseif T==FPGroup
+      G = T(GAP.Globals.PerfectGroup(GAP.Globals.IsSubgroupFpGroup,n,m))
+   else
+      throw(ArgumentError("Wrong type"))
+   end
+
+   return G
+end
+
+perfect_group(n::Int, m::Int) = perfect_group(FPGroup,n,m)
+
+"""
+    perfect_identification(G::Group)
+Return (`n`, `m`), where `G` = perfect_group(`n`,`m`).
+"""
+perfect_identification(G::GAPGroup) = GAP.gap_to_julia(GAP.Globals.PerfectIdentification(G.X))
+
+"""
+    number_perfect_groups(n::Int)
+Return the number of perfect groups of order `n`.
+"""
+number_perfect_groups(n::Int) = GAP.Globals.NumberPerfectGroups(n)
+
+"""
+    isperfect(G)
+Return whether `G` is a perfect group, i.e., equal to its derived subgroup.
+"""
+isperfect(G::GAPGroup) = GAP.Globals.IsPerfectGroup(G.X)
+
+
+###################################################################
+# Small groups
+###################################################################
+
+"""
+    small_group(n::Int, i::Int)
+Return the `i`-th group of order `n` in the catalogue of the GAP Small Groups Library. The group is given of type ``PcGroup`` if the group is solvable, ``PermGroup`` otherwise.
+"""
+function small_group(n::Int, m::Int)
+  @assert m<= number_small_groups(n) "There are less than $m groups of order $n."
+  G = GAP.Globals.SmallGroup(n, m)
+  T = _get_type(G)
+  return T(G)
+end
+
+"""
+    small_group_identification(G::Group)
+Return (`n`, `m`), where `G` = small_group(`n`,`m`).
+"""
+function small_group_identification(G::GAPGroup)
+  r = GAP.Globals.IdGroup(G.X)
+  res = GAP.gap_to_julia(GAP.gap_to_julia(r))
+  return (res[1], res[2])
+end
+
+"""
+    number_small_groups(n::Int)
+Return the number of small groups of order `n`.
+"""
+number_small_groups(n::Int) = GAP.Globals.NumberSmallGroups(n)
+
+
+###################################################################
+# AllSmallGroups
+###################################################################
+
+function find_index_function(f, isapg)
```

Comment:
This function and the code using it really should be documented (i.e. a comment should say what it is for).

Also, this really looks like it would be a good job for a dictionary, mapping these functions to the relevant tuples. That'd probably be faster, too.

Finally, is this code tested at all? I see no tests for `all_transitive_groups` etc. -- that's really bad. All new code should be tested, but esp. such complex code.